### ISSUE TITLE
Owner names not rendered in React Datasource editor

### ISSUE THREAD

**Issue by john-bodley** [apache/superset#11680]
### Expected results

The dataset owners should be rendered.

### Actual results

<img width="392" alt="Screen Shot 2020-11-12 at 2 37 42 PM" src="https://user-images.githubusercontent.com/4567245/99004984-d2ab6480-24f4-11eb-97c9-ec36900ea5cd.png">

#### How to reproduce the bug

1. Go to a chart.
2. Click on 'Edit Datasource'
3. Click on the 'SETTINGS' tab
4. Scroll to the bottom and and noticed that the owner names aren't rendered

### Environment

(please complete the following information):

- superset version: `master`
- python version: `3.7`
- node.js version: `12`

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by john-bodley** [comment]
@rusackas I believe this regression may be related to a recent change from Preset related to the Antd refactor.

**Comment by rusackas** [comment]
@kgabryje

**Comment by rusackas** [comment]
Thanks John, will investigate.

**Comment by junlincc** [comment]
<img width="871" alt="Screen Shot 2020-11-12 at 2 59 49 PM" src="https://user-images.githubusercontent.com/67837651/99006755-cffe3e80-24f7-11eb-8e35-7c68896bb13e.png">
@john-bodley hey John, im not able to reproduce it, do you have an idea which PR might cause it and when did you first noticed it?

**Comment by john-bodley** [comment]
@junlincc unsure. We believe this regression may have been introduced in one of [these changes](https://github.com/airbnb/incubator-superset/compare/release--0.92.2%E2%80%A6airbnb:release--0.93).

**Comment by rusackas** [comment]
> @junlincc unsure. We believe this regression may have been introduced in one of [these changes](https://github.com/airbnb/incubator-superset/compare/release--0.92.2%E2%80%A6airbnb:release--0.93).

I'm not able to repro this either yet. I would try a little `git bisect` action, but I need a failing state to do so. :/

**Comment by junlincc** [comment]
@john-bodley could you help narrow it down a bit 🙏 

> > @junlincc unsure. We believe this regression may have been introduced in one of [these changes](https://github.com/airbnb/incubator-superset/compare/release--0.92.2%E2%80%A6airbnb:release--0.93).
> 
> I'm not able to repro this either yet. I would try a little `git bisect` action, but I need a failing state to do so. :/

**Comment by john-bodley** [comment]
@junlincc I'm going to see if the issue is present in our internal weekly deploy which occurs on Wednesday. If it's resolved it's likely our last week branch cut did not contain the commit and I'll close the issue, otherwise I'll dig into this in more detail.

**Comment by john-bodley** [comment]
@junlincc and @rusackas I'm able to reproduce this from `master` using our production data. It seems like the issue is the fetched set of potential owners is quite restricted (the CRUD view fetches all users) and the prescribed owners are not within said set and thus the labels aren't rendered. 

The logic is defined [here](https://github.com/apache/incubator-superset/blob/55a3404b71232a1cfd6d978f666945cde37c99e2/superset-frontend/src/datasource/DatasourceEditor.jsx#L544-L552) and I only see one request to the `api/v1/dataset/related/owners` endpoint regardless of the typeahead state.

**Comment by junlincc** [comment]
@john-bodley thanks for the additional info we are looking into it!

**Comment by graceguo-supercat** [comment]
I found another JS bug for owners selector (besides the issue John just reported). this can be easily reproduced in open source master branch: 
- you add one more name in the owners list
- you remove all owners one by one. on removal the last one from owners list, you will see js error.

**Comment by kgabryje** [comment]
@junlincc Can you assign me please?

**Comment by kgabryje** [comment]
@john-bodley I think that this issue might have been introduced by https://github.com/apache/incubator-superset/pull/11221. I'm not able to verify it as I can't directly reproduce this issue with my test data. Is it possible that the new endpoint `api/v1/dataset/related/owners` returns fewer owners than the old `/users/api/read` and your data actually uses the missing users? Could you please verify it?
CC @lilykuang @junlincc

**Comment by john-bodley** [comment]
@dpgaspar do you know of any differences between the `/api/v1/dataset/related/owners` and `/users/api/read` API endpoints?

**Comment by nytai** [comment]
One difference is that `/api/v1/dataset/related/owners` is paginated (with `pageSize` and `pageIndex` params) while I'm guessing `/users/api/read` is not paginated. My guess is that airbnb is hitting this issue because they have >25 users which is the default page size. The `/api/v1/dataset/related/owners` also supports searching via `filter` param. 


My guess is that because the existing owners is not in the list of select options it's resulting in this weird rendering issue. We may want to consider just switching back to `/users/api/read` or trying to use the [pagianted select component](https://github.com/apache/incubator-superset/blob/fbe4a6622e0affd6568ba0d5c429dc6bec18f626/superset-frontend/src/components/Select/SupersetStyledSelect.tsx#L295) which may fix this issue

**Comment by kgabryje** [comment]
@nytai I think your guess is correct - I could reproduce John's bug by limiting a page size.
I wonder if paginated endpoint has any value for us in this case? We always need to load all options to make sure that we can display all selected values. Unless we implemented some logic "keep loading next pages until all saved values have a matching option" - though it doesn't sound like this optimization is worth the effort.

**Comment by etr2460** [comment]
@nytai @kgabryje @junlincc:

Are there any updates on resolving this? It seems like Tai had a few options for fixing the problem, is anything else blocking trying to fix it?

**Comment by nytai** [comment]
@kgabryje That's a great question. I have spend quite a bit of time working around this (adding `PaginatedSelect`) and have often questioned the value of paginating these payloads. @dpgaspar do you have any thoughts around removing pagination for these endpoints?

**Comment by nytai** [comment]
We could also send the request with `page_size=2000` or something large enough to solve this issue for most practical cases.

**Comment by dpgaspar** [comment]
We will probably go with `page_size=2000`, and address this properly next. A possible solution is to create a small feature on the generic related endpoint to support something like:
```
{"filter": "<text to filter>", "ids": [1,2,4,n], page:X, page_size:Y}
```
Where the endpoint would get filtered paginated data and make sure the requested ids were included on the response also. At a probable cost of a further query but still more efficient then sending back a very large payload

**Comment by zuzana-vej** [comment]
FYI, the above linked PR didn't resolve the issue.

See owners on "Edit Dataset" modal in Explore:
![Screen Shot 2021-03-01 at 6 14 12 PM](https://user-images.githubusercontent.com/61221714/109586495-277de580-7aba-11eb-8d3b-078d02b69232.png)

See owners on the Data --> Dataset page for same dataset
![Screen Shot 2021-03-01 at 6 14 59 PM](https://user-images.githubusercontent.com/61221714/109586482-25b42200-7aba-11eb-959b-3b6350653bd1.png)

**Comment by codenamelxl** [comment]
It also does not trigger Async load for user name when typing to add new user in the Owner. No request to /dataset/related/owner was made on typing to the Owner in Datasource Editor.
Behavior for Chart and Dashboard is it trigger /related/owner  on every keytype.

**Comment by etr2460** [comment]
Hey @amitmiran137 , is Neilsen still planning on working on this? Thanks!

**Comment by etr2460** [comment]
@michael-s-molina do you think this is something that your async select work could help with? This seems to be a regression dating back to the original Antd refactor

**Comment by michael-s-molina** [comment]
> @michael-s-molina do you think this is something that your async select work could help with? This seems to be a regression dating back to the original Antd refactor

@etr2460 Yes. I'm tagging @geido here because he worked on this migration and has the full context.

**Comment by geido** [comment]
@etr2460 I don't think the current implementation would work. You can check the code in this PR https://github.com/apache/superset/pull/16609. Basically, we would need a refactor of the `SelectAsyncControl` to work with the pagination. That PR is still not merged so this might be a good opportunity to think about doing it depending on prioritization. CC @junlincc

**Comment by nytai** [comment]
there is a workaround to use the Legacy Datasource Editor, but this bug does block adding new owners beyond the 10 that are initially fetched. This should be fixed before we fully deprecate the legacy FAB views

**Comment by michael-s-molina** [comment]
@junlincc @nytai Can you add some context about the reopening of this issue? I just checked it on master and it seems to be working as intended. I simulated a page with 10 results.

https://user-images.githubusercontent.com/70410625/144238905-286e81a8-8321-4e05-84dd-982d2d02526f.mov

**Comment by nytai** [comment]
This is still an issue. @michael-s-molina please try with > 25 results

**Comment by nytai** [comment]
The call to fetch owners is `https://superset.aiml.apple.com/api/v1/dataset/related/owners` which is falling back to the FAB default limit of 20. Other Select components have pagination enabled

**Comment by nytai** [comment]
this appears to be fixed in the latest version (1.4.0)?

**Comment by nytai** [comment]
Yup, looks like it was fixed by https://github.com/apache/superset/pull/17063
